gh-91248: Add PyFrame_GetVar() function (#95712)
python/cpython: Doc/c-api/frame.rst

   .. versionadded:: 3.11

.. c:function:: PyObject* PyFrame_GetVar(PyFrameObject *frame, PyObject *name)

   Get the variable *name* of *frame*.

   * Return a :term:`strong reference` to the variable value on success.
   * Raise an exception and return ``NULL`` on error.

>    * Return ``NULL`` if the variable does not exist.

Reviewer: That sounds like a confusing API that we don't need more of.

   .. versionadded:: 3.12

.. c:function:: PyObject* PyFrame_GetVarString(PyFrameObject *frame, const char *name)

>    Similar to :c:func:`PyFrame_GetVar`, but *name* bytes string is decoded from
>    UTF-8.

Reviewer: ```suggestion
   Similar to :c:func:`PyFrame_GetVar`, but the variable name is a C string encoded in
   UTF-8.
```
